cpp/CMakeLists.txt

```diff
@@ -706,8 +692,11 @@ endif(HT_DEFAULT_ALLOCATOR)
 ###################################################################################################
 # - link libraries --------------------------------------------------------------------------------
 
+# Get all the symbols from the Arrow CUDA Library for Cython
+set(ARROW_CUDA_LIB_LINK -Wl,--whole-archive ${ARROW_CUDA_LIB} -Wl,--no-whole-archive)
+
```

Comment:
This was a hack for vendored Cython code since there was a PyArrow gpu enabled package
```suggestion
```